src/main/java/software/bernie/techarium/block/pipe/PipeBlock.java

```diff
@@ -0,0 +1,192 @@
+import net.minecraft.block.BlockState;
+import net.minecraft.block.material.Material;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.ActionResultType;
+import net.minecraft.util.Direction;
+import net.minecraft.util.Hand;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.BlockRayTraceResult;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.World;
+import net.minecraft.world.server.ServerWorld;
+import net.minecraftforge.common.util.LazyOptional;
+import org.jetbrains.annotations.Nullable;
+import software.bernie.techarium.item.PipeItem;
+import software.bernie.techarium.pipes.capability.IPipeNetworkManagerCapability;
+import software.bernie.techarium.pipes.capability.PipeNetworkManagerCapability;
+import software.bernie.techarium.pipes.capability.PipeType;
+import software.bernie.techarium.registry.ItemRegistry;
+import software.bernie.techarium.tile.pipe.PipeTile;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public class PipeBlock extends Block {
+    public PipeBlock() {
+        super(AbstractBlock.Properties.create(Material.ROCK));
+    }
+
+    @Override
+    public boolean hasTileEntity(BlockState state) {
+        return true;
+    }
+
+    @Nullable
+    @Override
+    public TileEntity createTileEntity(BlockState state, IBlockReader world) {
+        return new PipeTile();
+    }
+
+    @Override
+    public void onBlockPlacedBy(World worldIn, BlockPos pos, BlockState state, @Nullable LivingEntity placer, ItemStack stack) {
+        if (worldIn.isRemote)
+            return;
+        if (placer != null && placer.isSneaking()) {
+            TileEntity te = worldIn.getTileEntity(pos);
+            if (te instanceof PipeTile) {
+                ((PipeTile) te).isInput = true;
+            }
+        }
+        if (!worldIn.isRemote())
+            handlePlace(state, worldIn, pos, stack);
+    }
+
+    @Override
+    public ActionResultType onBlockActivated(BlockState state, World worldIn, BlockPos pos, PlayerEntity player, Hand handIn, BlockRayTraceResult hit) {
+        ItemStack activatedWith =  player.getHeldItem(handIn);
+        if (activatedWith.getItem() instanceof PipeItem && !worldIn.isRemote && handlePlace(state, worldIn, pos, activatedWith))
+            return ActionResultType.CONSUME;
+        return super.onBlockActivated(state, worldIn, pos, player, handIn, hit);
+    }
+
+    @Override
+    public ItemStack getItem(IBlockReader worldIn, BlockPos pos, BlockState state) {
+        TileEntity te = worldIn.getTileEntity(pos);
+        if (te instanceof PipeTile) {
+            PipeTile pipe = ((PipeTile)te);
+            if (pipe.isType(PipeType.ITEM))
```

Comment:
Preferably there'd be a registry for pipetypes so that other mods could register their own (like AE2, etc.)

Reply:
The ModelBakeEvent would need to know how to handle more than 3 and I'm not sure if it is a good idea to dynamically add more blockstateproperties. But I can keep it in mind, designing future code